tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java

```diff
@@ -369,61 +369,62 @@ else if (col.getColumnInfo().isPrimaryKey() && tableInfo.isPkHandle()) {
       }
 
       dagRequestBuilder.addExecutors(executorBuilder.setTblScan(tblScanBuilder));
-    }
 
-    // clear executorBuilder
```

Comment:
changes in this file, just move selection, limit executor building process into table scan.